### DISCUSSION TITLE
enthalpy vs. temperature calculation

### DISCUSSION THREAD

**Discussion by petrvokac** [pycalphad/pycalphad#414] [General]
[feo-h-pycalphad-vs-oc6.tar.gz](https://github.com/pycalphad/pycalphad/files/8698631/feo-h-pycalphad-vs-oc6.tar.gz)

Hello,
please find in the attached archive an example of enthalpy vs. temperature calculation.
I compared results with OpenCalphad6. The are significant differences for few values. OC6 also does not calculate the whole temperature range in one calculation. However OC6 reports an error therefore it can handled. My question is: how can I identify wrong results in pycalphad calculation? 
Thank you,
Petr

**Comment by richardotis** [comment]
This is the result I get on `0.10.0`:
![Fe-O Enthalpy 0.10.0](https://user-images.githubusercontent.com/6405510/168628660-f279099b-d630-49e9-bfe9-840572a140aa.png)

This is what I get on the latest `develop` branch (`f8e3951`):
![Fe-O Enthalpy develop](https://user-images.githubusercontent.com/6405510/168630119-e5358e7a-d420-45c1-aa9b-29d1c31a3ac2.png)


I'm not sure about the divergence at the intermediate temperature on 0.10.0, but the development version looks like it's doing the correct thing now. The reason for the divergence at 290 K is because pycalphad does not extrapolate beyond the lower and upper temperature limits specified in a model parameter: https://github.com/pycalphad/pycalphad/issues/306#issuecomment-786249501

We do not recommend performing calculations below 300 K unless the database you are using was specifically designed for it. If you happen to know that a model expression is valid beyond its lower or upper limit, you can edit the database to use an artificial lower limit (1 K) or upper limit (10000 K).

This is what I get on `develop` if I modify your calculation to start at 300 K instead:
![Fe-O Enthalpy develop 300 Kelvin start](https://user-images.githubusercontent.com/6405510/168630678-b326f7b9-d007-41aa-b1a5-8319fe2e7aed.png)

**Reply by petrvokac** [reply]
Thank you for explanation! I should be more careful and always look at temperature limits of the database. My reason for choice 290K as lower limit is to have 298.15K in the tabulated range. I am looking forward for the updated version.

**Reply by richardotis** [reply]
Starting your lower limit at exactly 298.15 should work.
